webpack/assets/javascripts/react_app/routes/ForemanRoute.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { layoutPropTypes } from '../components/Layout/Layout';
+import RouteWithLayout from './RouteWithLayout';
+import { noop } from '../common/helpers';
+
+const ForemanRoute = ({
+  render,
+  layout,
+  skipLayout,
+  beforeRender,
+  path,
+  ...routeProps
+}) => {
+  if (skipLayout) {
+    return (
+      <Route
+        path={path}
+        key={path}
+        {...routeProps}
+        render={renderProps => {
+          beforeRender(renderProps);
+
+          return render(renderProps);
+        }}
+      />
+    );
+  }
+  return (
+    <RouteWithLayout
+      path={path}
+      key={path}
+      render={renderProps => {
+        beforeRender(renderProps);
+
+        return render(renderProps);
+      }}
+      layout={layout}
+    />
+  );
+};
+
+export default ForemanRoute;
+
+delete layoutPropTypes.history;
```

Comment:
This looks a bit strange to me,
if you want to remove some of the layout `proptypes`,
it might be better to use lodash's `omit` method in the layout propTypes definition:
```
layout: PropTypes.shape(omit(layoutPropTypes, 'history')).isRequired,
```

another option is to use patternfly's excludeKeys method on the props.